backend/src/connector/mod.rs

```diff
@@ -1 +1,267 @@
+use self::{
+    api::{APIError, ApiNoContext},
+    client::Client,
+    context::ClientContext,
+};
+use crate::{
+    models::{
+        bob::NodeName,
+        shared::{BobConnectionData, Hostname, RequestTimeout, XSpanIdString},
+    },
+    prelude::*,
+};
+use api::ContextWrapperExt;
+use axum::headers::{authorization::Basic, Authorization};
+use error_stack::ResultExt;
+use frunk::hlist;
+use hyper::StatusCode;
+use std::{collections::HashMap, sync::Arc};
+use thiserror::Error;
 
+pub mod api;
+pub mod client;
+pub mod context;
+pub mod dto;
+
+pub type ApiInterface = dyn ApiNoContext<ClientContext> + Send + Sync;
```

Comment:
Why it is a dynamic type? We have a single Client implementation, so it can be a specific type